File: Go/autogram-sha256.go
package main

import "fmt"
import "crypto/sha256"

func main(){

	numbers_dict := map[string]string{
		"zero":"0",
		"one":"1",
		"two":"2",
		"three":"3",
		"four":"4",
		"five":"5",
		"six":"6",
		"seven":"7",
		"eight":"8",
		"nine":"9" }
	
	alphabets := []string{"a","b","c","d","e","f"}
	numbers_words := []string{"zero","one","two","three","four","five","six","seven","eight","nine"}

	for _, a1 := range alphabets {
		for _, a2 := range alphabets {
			for _, n1 := range numbers_words {
				for _, n2 := range numbers_words {
					for _, n3 := range numbers_words {
						for _, n4 := range numbers_words {
							for _, n5 := range numbers_words {

								autogram := numbers_dict[n1] + numbers_dict[n2] + numbers_dict[n3] + a1 + numbers_dict[n4] + a2 + numbers_dict[n5]

								sentence := fmt.Sprintf("The SHA256 for this sentence begins with: %s, %s, %s, %s, %s, %s and %s.", n1, n2, n3, a1, n4, a2, n5)

								// Convert the input into a byte slice
								sentence_byte := []byte(sentence)

								// Create a new SHA-256 hash object
								hasher := sha256.New()

								// Compute the hash
								hasher.Write(sentence_byte)

								// Calculate the hash of the input sentence and store it as a byte slice
								hashBytes := hasher.Sum(nil)

								// Get the hexadecimal representation of the hash
								sentence_hash := fmt.Sprintf("%x", hashBytes)

								if autogram == sentence_hash[:7] {
									fmt.Println(sentence)
									fmt.Println(sentence_hash)
								}
								
							}
						}
					}
				}
			}
		}
	}
}

File: C#/autogram-sha256.cs
using System.Security.Cryptography;
using System.Text;

public class Autogram_SHA256{
    
    public static void Autogram_SHA256SUM(){

        Dictionary<string, string> numbers_dict = new Dictionary<string, string>{
            {"zero", "0"},
            {"one", "1"},
            {"two", "2"},
            {"three", "3"},
            {"four", "4"},
            {"five", "5"},
            {"six", "6"},
            {"seven", "7"},
            {"eight", "8"},
            {"nine", "9"}
        };

        string[] alphabets = {"a","b","c","d","e","f"};
        string[] numbers_words = {"zero","one","two","three","four","five","six","seven","eight","nine"};

        foreach (string a1 in alphabets) {
            foreach (string a2 in alphabets) {
                foreach (string n1 in numbers_words) {
                    foreach (string n2 in numbers_words) {
                        foreach (string n3 in numbers_words) {
                            foreach (string n4 in numbers_words) {
                                foreach (string n5 in numbers_words) {

                                    string autogram = numbers_dict[n1] + numbers_dict[n2] + numbers_dict[n3] + a1 + numbers_dict[n4] + a2 + numbers_dict[n5];

                                    string sentence = $"The SHA256 for this sentence begins with: {n1}, {n2}, {n3}, {a1}, {n4}, {a2} and {n5}.";

                                    // Convert the string to bytes (UTF-8 encoding is used here)
                                    byte[] sentence_byte = Encoding.UTF8.GetBytes(sentence);

                                    // Create an instance of the SHA-256 algorithm
                                    using SHA256 sha256 = SHA256.Create();

                                    // Compute the hash
                                    byte[] hashBytes = sha256.ComputeHash(sentence_byte);

                                    // Get the hexadecimal representation of the hash
                                    string sentence_hash = BitConverter.ToString(hashBytes).Replace("-", "").ToLower();

                                    if (autogram == sentence_hash.Substring(0, 7)){
                                        Console.WriteLine(sentence);
                                        Console.WriteLine(sentence_hash);
                                    }

                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
